DatadogRUM/Sources/Instrumentation/WatchdogTerminations/WatchdogTerminationChecker.swift

```diff
@@ -0,0 +1,118 @@
+                    guard let self = self else {
+                        completion(false)
+                        return
+                    }
+                    completion(self.isWatchdogTermination(launch: launch, from: previous, to: current))
+                }
+            }
+        } catch let error {
+            DD.logger.error("Failed to check if Watchdog Termination occurred", error: error)
+            completion(false)
+            throw error
+        }
+    }
+
+    /// Checks if the app was terminated by Watchdog.
+    /// - Parameters:
+    ///  - launch: The launch report containing information about the app launch.
+    ///  - previous: The previous app state stored in the data store from the last app session.
+    ///  - current: The current app state of the app.
+    func isWatchdogTermination(
+        launch: LaunchReport,
+        from previous: WatchdogTerminationAppState?,
+        to current: WatchdogTerminationAppState
+    ) -> Bool {
+        DD.logger.debug(launch.debugDescription)
+        DD.logger.debug(previous.debugDescription)
+        DD.logger.debug(current.debugDescription)
+
+        guard let previous = previous else {
+            return false
+        }
+
+        // Watchdog Termination detection doesn't work on simulators.
+        guard deviceInfo.isSimulator == false else {
+            return false
+        }
+
+        // When the app is running in debug mode, we can't reliably tell if it was a Watchdog Termination or not.
+        guard previous.isDebugging == false else {
+           return false
+        }
+
+        // Is the app version different than the last time the app was opened?
+        guard previous.appVersion == current.appVersion else {
+            return false
+        }
+
+        // Is there a crash from the last time the app was opened?
+        guard launch.didCrash == false else {
+            return false
+        }
+
+        // Did we receive a termination call the last time the app was opened?
+        guard previous.wasTerminated == false else {
+            return false
+        }
+
+        // Is the OS version different than the last time the app was opened?
+        guard previous.osVersion == current.osVersion else {
+            return false
+        }
+
+        // Was the system rebooted since the last time the app was opened?
+        guard previous.systemBootTime == current.systemBootTime else {
+            return false
+        }
+
+        // This value can change when installing test builds using Xcode or when installing an app
+        // on a device using ad-hoc distribution.
+        guard previous.vendorId == current.vendorId else {
+            return false
+        }
+
+        // Was the app in foreground/active ?
+        // If the app was in background we can't reliably tell if it was a Watchdog Termination or not.
+        guard previous.isActive else {
+            return false
+        }
+
+        return true
```

Comment:
question/ I wonder if we shouldn't consider [ProcessIdentifier](https://github.com/DataDog/dd-sdk-ios/blob/dc16eac3e8cb2644573edb0cadfa052816b1d052/DatadogRUM/Sources/Instrumentation/AppHangs/ProcessIdentifier.swift#L9-L19) as part of this heuristic, same as [we do it in Fatal App Hangs monitoring](https://github.com/DataDog/dd-sdk-ios/blob/dc16eac3e8cb2644573edb0cadfa052816b1d052/DatadogRUM/Sources/Instrumentation/AppHangs/FatalAppHangsHandler.swift#L69-L71). Consider scenario:
- 1/ start app process
- 2/ then start SDK (RUM + Crash Reporting)
- 3/ then stop SDK
- 4/ then start SDK again

wouldn't current heuristic flag 3 → 4 as watchdog termination? To avoid such false-positive in fatal hangs monitoring, we check if the global `currentProcessID = UUID()` has changed from previous SDK instance to the new one.